### PR TITLE
Fix error message when trying to use using-static with an invalid type

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/68546

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
@jcouv ptal

File: src/Compilers/CSharp/Portable/CSharpResources.resx [hunk 1]
```diff
@@ -7511,4 +7523,7 @@ To remove the warning, you can use /reference instead (set the Embed Interop Typ
   <data name="ERR_UnsupportedPrimaryConstructorParameterCapturingRefAny" xml:space="preserve">
     <value>Cannot use primary constructor parameter of type '{0}' inside an instance member</value>
   </data>
+  <data name="ERR_BadUsingStaticType" xml:space="preserve">
+    <value>'{0}' type is not valid for 'using static'. Only a class, struct, interface, enum, delegate or namespace can be used.</value>
```
Comment (Youssef1313) [inline src/Compilers/CSharp/Portable/CSharpResources.resx:7527]:
```suggestion
    <value>'{0}' type is not valid for 'using static'. Only a class, struct, interface, enum, delegate, or namespace can be used.</value>
```

Comment (Youssef1313) [reply src/Compilers/CSharp/Portable/CSharpResources.resx:7527]: - Oxford comma is commonly used, I think. e.g: https://github.com/dotnet/roslyn/blob/591154398bcbcd5793908d62156f7f2366b23426/src/Compilers/CSharp/Portable/CSharpResources.resx#L712
- *Most* errors don't end with period.

```suggestion
    <value>'{0}' type is not valid for 'using static'. Only a class, struct, interface, enum, delegate or namespace can be used</value>
```

**Comment by CyrusNajmabadi** [comment]
i see a massive amount of errors with ending periods.  I'm fine keeping as is.  My pref would be that if we care about ending periods we apply it to everything and are consistent everywhere.

**Comment by CyrusNajmabadi** [comment]
I changed the comma.  I see both styles in the compiler, but it does look like a lot more use the oxford comma.

File: src/Compilers/CSharp/Portable/CSharpResources.resx [hunk 2]
```diff
@@ -7511,4 +7523,7 @@ To remove the warning, you can use /reference instead (set the Embed Interop Typ
   <data name="ERR_UnsupportedPrimaryConstructorParameterCapturingRefAny" xml:space="preserve">
     <value>Cannot use primary constructor parameter of type '{0}' inside an instance member</value>
   </data>
+  <data name="ERR_BadUsingStaticType" xml:space="preserve">
+    <value>'{0}' type is not valid for 'using static'. Only a class, struct, interface, enum, delegate, or namespace can be used.</value>
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Portable/CSharpResources.resx:7527]: >'{0}' [](http://example.com/codeflow?start=11&length=5)

Not starting a sentence with a capital letter might look strange.

File: src/Compilers/CSharp/Test/Symbol/Symbols/Source/UsingAliasTests.cs [hunk 3]
```diff
@@ -969,5 +970,104 @@ void M()
...
+                    using static A;
+                }
+                """, options: TestOptions.UnsafeDebugDll).VerifyDiagnostics(
+                    // (5,5): hidden CS8019: Unnecessary using directive.
+                    //     using static A;
+                    Diagnostic(ErrorCode.HDN_UnusedUsingDirective, "using static A;").WithLocation(5, 5),
+                    // (5,18): error CS9137: 'pointer' type is not valid for 'using static'. Only a class, struct, interface, enum, delegate or namespace can be used.
+                    //     using static A;
+                    Diagnostic(ErrorCode.ERR_BadUsingStaticType, "A").WithArguments("pointer").WithLocation(5, 18),
+                    // (5,18): error CS0214: Pointers and fixed size buffers may only be used in an unsafe context
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Test/Symbol/Symbols/Source/UsingAliasTests.cs:1010]: >// (5,18): error CS0214: Pointers and fixed size buffers may only be used in an unsafe context [](http://example.com/codeflow?start=20&length=94)

Consider suppressing this error to reduce the noise.

File: src/Compilers/CSharp/Test/Symbol/Symbols/Source/UsingAliasTests.cs [hunk 4]
```diff
@@ -969,5 +970,104 @@ void M()
...
+                    using static A;
+                }
+                """, options: TestOptions.UnsafeDebugDll).VerifyDiagnostics(
+                    // (5,5): hidden CS8019: Unnecessary using directive.
+                    //     using static A;
+                    Diagnostic(ErrorCode.HDN_UnusedUsingDirective, "using static A;").WithLocation(5, 5),
+                    // (5,18): error CS9137: 'function pointer' type is not valid for 'using static'. Only a class, struct, interface, enum, delegate or namespace can be used.
+                    //     using static A;
+                    Diagnostic(ErrorCode.ERR_BadUsingStaticType, "A").WithArguments("function pointer").WithLocation(5, 18),
+                    // (5,18): error CS0214: Pointers and fixed size buffers may only be used in an unsafe context
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Test/Symbol/Symbols/Source/UsingAliasTests.cs:1032]: >// (5,18): error CS0214: Pointers and fixed size buffers may only be used in an unsafe context [](http://example.com/codeflow?start=20&length=94)

Here as well

**Review by AlekseyTs — APPROVED** [review]
LGTM (commit 3)

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-compiler For the second review

**Comment by sharwell** [comment]
> Most errors don't end with period.

@Youssef1313 I agree, and for errors fitting the pattern we should strive to make all new messages consistent with the guidelines. Note that the exception to this case is an error with more than one sentence, where both will end with a period. This particular diagnostic fits the exception criteria (several others under nullable reference types do as well).

File: src/Compilers/CSharp/Portable/CSharpResources.resx (hunk 2 above)
Comment (CyrusNajmabadi) [reply src/Compilers/CSharp/Portable/CSharpResources.resx:7527]: Personally, i found the messages clear :)

File: src/Compilers/CSharp/Test/Symbol/Symbols/Source/UsingAliasTests.cs (hunk 3 above)
Comment (CyrusNajmabadi) [reply src/Compilers/CSharp/Test/Symbol/Symbols/Source/UsingAliasTests.cs:1010]: sure.  let me see how possible that is.

File: src/Compilers/CSharp/Portable/CSharpResources.resx (hunk 2 above)
Comment (333fred) [reply src/Compilers/CSharp/Portable/CSharpResources.resx:7527]: As long as it's a `'`, I think it'll be fine.

**Review by 333fred — APPROVED** [review]

**Pull request merged**